src/Stream/src/StreamBase/Binder/PartitionHandler.cs

```diff
@@ -37,51 +39,15 @@ public class PartitionHandler
 
         public int PartitionCount { get; set; }
 
-        public IExpressionParser ExpressionParser
-        {
-            get
-            {
-                if (_expressionParser == null)
-                {
-                    _expressionParser = IntegrationContextUtils.GetExpressionParser(_applicationContext);
-                }
-
-                return _expressionParser;
-            }
-
-            set
-            {
-                _expressionParser = value;
-            }
-        }
-
-        public IEvaluationContext EvaluationContext
-        {
-            get
-            {
-                if (_evaluationContext == null)
-                {
-                    _evaluationContext = IntegrationContextUtils.GetEvaluationContext(_applicationContext);
-                }
-
-                return _evaluationContext;
-            }
-
-            set
-            {
-                _evaluationContext = value;
-            }
-        }
-
         public int DeterminePartition(IMessage message)
         {
             var key = ExtractKey(message);
 
             int partition;
-            if (!string.IsNullOrEmpty(_producerOptions.PartitionSelectorExpression) && ExpressionParser != null)
+            if (!string.IsNullOrEmpty(_producerOptions.PartitionSelectorExpression) && _expressionParser != null)
             {
-                var expr = ExpressionParser.ParseExpression(_producerOptions.PartitionSelectorExpression);
-                partition = expr.GetValue<int>(EvaluationContext, key);
+                var expr = _expressionParser.ParseExpression(_producerOptions.PartitionSelectorExpression);
+                partition = expr.GetValue<int>(_evaluationContext ?? new StandardEvaluationContext(), key);
```

Comment:
Would it make sense to move `?? new StandardEvaluationContext()` up to the constructor so that we can know that `_evaluationContext` is never null?